### ISSUE TITLE
Calls to trait _constructors_ emmited

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#427]
```
trait A
object B extends A
```

```
  public B$();
    Code:
       0: aload_0
       1: invokespecial #21                 // Method java/lang/Object."<init>":()V
       4: aload_0
       5: putstatic     #13                 // Field MODULE$:LB$;
       8: aload_0
       9: invokespecial #22                 // Method A."<init>":()V
      12: return
```

Trees seem to originate from `constructors` phase

**Comment by DarkDimius** [comment]
Fails jvm verification.
```
Exception in thread "main" java.lang.VerifyError: Bad operand type when invoking <init>
Exception Details:
  Location:
    B$.<init>()V @9: invokespecial
  Reason:
    Invalid type: 'B$' (current frame, stack[0])
  Current Frame:
    bci: @9
    flags: { }
    locals: { 'B$' }
    stack: { 'B$' }
  Bytecode:
    0x0000000: 2ab7 0015 2ab3 000d 2ab7 0016 b1

	at B.main(A.scala)
```

**Comment by DarkDimius** [comment]
fixed in #428
